### DISCUSSION TITLE
transcoding video files - how?

### DISCUSSION THREAD

**Discussion by x3nb63** [concourse/concourse#8221] [Help & Support] [answered]
I have large video files in S3 buckets and each needs to be transcoded. Tens of GB in size for each - so download/upload is expensive.

For the transcoding part I already have a docker image that expects a single file in a input volume and will write the result to an output volume  (while accepting some control via environment). 

Now the obvious idea is to make a pipeline somehow find each file and issue a transcoding task running that image, one for each file found.

How would you approach that pipeline?

I see there is this s3-resource that can get and put things from/to buckets ... but I can not have it get the full bucket cause its way too big for my workers. I want to pick only a single file at once + transcode + upload to another bucket *while* ensuring to not forget a single one and to not process a file twice ...

I certainly dont want to configure a resource for each single file.

It sounds like `across step` to me but that would need to be filled with (a long) list of files + paths ...  or self-discover the list? ...

Do I need to write my own resource with finding and tracking smartness in it?

**Comment by taylorsilva** [comment, marked as answer]
The S3 resource would not grab the entire bucket, it would grab one file at a time. Though I don't think the current s3 resource would work well for your case unless you're naming your video files with semver. You'll probably need to make your own version of the s3 resource that grab the latest file added to the bucket, or something similar to that based on how I'm understanding your problem.

So a custom s3 resource that pulls in one video per build of a job. That's probably the solution I would try building towards.

**Reply by x3nb63** [reply]
yes, I crafted two designs by now:   one with a git repo as brain (commit files with path lists + trigger on these) and another by purely using a bucket with in/out directory tree. Both ended up with the need to put custom things into the s3-resource.
